frontend/src/components/Sidebar/HeatLegendEdit.tsx

```diff
@@ -22,143 +29,217 @@ import {
+         (presetList: HeatmapLegend[]) => {
+           presetList.forEach((legend) => {
+             if (legend.isNormalized) {
+               legend.steps.forEach((step) => {
+                 //set step to normalized values
+                 step.value = step.value / legend.steps[legend.steps.length - 1].value;
+
+
+               });
+             }
+           });
+           //create default legends
+           const defaultLegends: HeatmapLegend[] = [];
+           const stepCount = theme.custom.scenarios[0].length - 1;
+           for (let i = 0; i < theme.custom.scenarios.length; i++) {
+
+             const steps = [];
+             for (let j = 0; j < stepCount; j++) {
+               steps.push({color: theme.custom.scenarios[i][stepCount - 1 - j], value: j / (stepCount - 1)});
+             }
+             defaultLegends.push({name: 'Default', isNormalized: true, steps});
+           }
+           dispatch(setDefaultLegends({legends: defaultLegends}));
+
+           //add default preset to the front of the presetList
+           presetList.unshift(defaultLegends[0]);
+           // fill presets state with list
+           dispatch(setHeatmapLegends({legends: presetList}));
+           // select default legend
+           dispatch(selectHeatmapLegend({legend: defaultLegends[0]}));
+         },
+         // Reject Promise
+         () => {
+           console.warn('Did not receive proper heatmap legend presets');
+         }
+         );
+       // this init should only run once on first render
+     }
+     // eslint-disable-next-line react-hooks/exhaustive-deps
+   }, []);
+
+   useEffect(() => {
+     if (activeScenario) {
+       //change the default Legend according to the active Scenario
+       dispatch(selectDefaultLegend({selectedScenario: activeScenario - 1}));
+     }
+   }, [activeScenario, dispatch]);
+
+   return (
+     <>
+     <Tooltip title={t('heatlegend.edit').toString()} placement='right' arrow>
+
+     <IconButton
+     color={'primary'}
+     onClick={() => setHeatLegendEditOpen(true)}
+     aria-label={t('heatlegend.edit')}
+     size='small'
+     sx={{padding: theme.spacing(0), marginBottom: theme.spacing(1)}}
+     >
+
+     <EditIcon />
+     </IconButton>
+     </Tooltip>
+     <Dialog maxWidth='lg' fullWidth={true} open={heatLegendEditOpen} onClose={() => setHeatLegendEditOpen(false)}>
+     <Box
+     sx={{
+       padding: theme.spacing(4),
+       background: theme.palette.background.paper,
+     }}
+     >
+    
+        
+    
+     <span style={{float:'right'}}>
+
+<FormGroup>
+  <FormControlLabel control= {<Switch checked= {isOn} onChange ={()=> setIsOn(!isOn)}/>}label="Show more" labelPlacement='start'/>
+</FormGroup>
+
+     {/* <Toggle isOn={isOn} handleChange={() => setIsOn(!isOn)} /> */}
+    {/*  <Switch {...label} checked={isOn} onChange={() =>setIsOn(!isOn)}/> */}
+     </span>
+   
+
+     <FormControl fullWidth sx={{marginBottom: theme.spacing(3)}}>
+
+     {!isOn  ? ( 
```

Comment:
Do we need to replicate the whole JSX for the two cases?
Can't we check isOn and shorten the list if necessary, then use either list for the map?
This way you don't need to have this large duplicate JSX.